src/collector.ts

```diff
@@ -88,22 +91,62 @@ class NaivePyParser {
 class ReqDependencyCollector implements IDependencyCollector {
     constructor(public classes: Array<string> = ["dependencies"]) {}
 
-    async collect(contents: string): Promise<Array<IDependency>> {
+    async collect(contents: string, manifestFile: string): Promise<Array<IDependency>> {
         let parser = new NaivePyParser(contents);
         return parser.parse();
     }
 
 }
 
 class NaiveGomodParser {
-    constructor(contents: string) {
-        this.dependencies = NaiveGomodParser.parseDependencies(contents);
+    constructor(contents: string, manifestFile: string) {
+        this.dependencies = NaiveGomodParser.parseDependencies(contents, manifestFile);
     }
 
     dependencies: Array<IDependency>;
 
-    static parseDependencies(contents:string): Array<IDependency> {
+    static parseDependencies(contents:string, manifestFile: string): Array<IDependency> {
         const gomod = contents.split("\n");
+
+        const vscodeRootpath = manifestFile.replace("file://", "").replace("/go.mod", "")
+        const targetDir = paths.join(vscodeRootpath, 'target');
+        const goImportsFilePath = paths.join(targetDir, 'goimports.txt');
+        
+        // Create target folder if not exists.
+        if (!fs.existsSync(targetDir)) {
+            fs.mkdirSync(targetDir);
+        }
+
+        const cmd: string = [
+            `cd`,
+            `"${vscodeRootpath}" &&`,
+            `go`,
+            `list`,
+            `-f`,
+            `"{{.Imports}}"`,
+            `./...`,
+            `>`,
+            `"${goImportsFilePath}"`,
+        ].join(' ');
```

Comment:
Also I would suggest to use the following command to get list of imports.,
```
go list -f '{{ join .Imports "\n" }}' ./... 
```

Once you get list of imports, then you can deduplicate it using `Set`(like `go list -f '{{ join .Imports "\n" }}' ./... | sort | uniq `) and join it with modules listed in `go.mod`.

Reply:
'sort' and 'uniq' I am avoiding as this might not be available on "Windows" host. Its better to stick to go commands for information.

Reply:
I didn’t suggest to use uniq and sort commands, the ask is to do the equivalent in typescript your time reduce the overall items.

Reply:
This is done using Set()